jina/peapods/peas/__init__.py

```diff
@@ -194,9 +195,13 @@ def terminate(self):
         This method calls :meth:`terminate` in :class:`threading.Thread` or :class:`multiprocesssing.Process`.
         """
         if hasattr(self.worker, 'terminate'):
-            self.logger.debug(f' terminating the runtime process')
+            self.logger.debug(f'terminating the runtime process')
             self.worker.terminate()
             self.logger.debug(f' runtime process properly terminated')
+        else:
+            self.logger.debug(f'canceling the runtime thread')
+            self.runtime_cls.cancel(cancel_event=self.cancel_event)
```

Comment:
now all the runtimes should be equivalent and simply set cancel event  and thats end of the story? or is it still good to have it in the runtime?

Reply:
true, setting the cancel event directly would also work. I have no strong opinion here, but letting the runtime decide that looks a bit cleaner to me?

Reply:
yes, it can be considered cleaner, but maybe it is more readable if done here

Reply:
sure, changed it